src/java/org/apache/cassandra/db/rows/UnfilteredSerializer.java

```diff
@@ -587,7 +588,10 @@ public Row deserializeRowBody(DataInputPlus in,
             {
                 long timestamp = header.readTimestamp(in);
                 int ttl = hasTTL ? header.readTTL(in) : LivenessInfo.NO_TTL;
-                int localDeletionTime = hasTTL ? header.readLocalDeletionTime(in) : LivenessInfo.NO_EXPIRATION_TIME;
+                long localDeletionTime = hasTTL ? header.readLocalDeletionTime(in) : LivenessInfo.NO_EXPIRATION_TIME;
+                if (localDeletionTime < 0)
+                    localDeletionTime = helper.version < MessagingService.VERSION_42 ? Cell.LEGACY_OVERFLOWED_DELETION_TIME : Cell.deletionTimeUnsignedIntegerToLong((int) localDeletionTime);
```

Comment:
Nit: split the long line
```suggestion
                    localDeletionTime = helper.version < MessagingService.VERSION_42
                                        ? Cell.LEGACY_OVERFLOWED_DELETION_TIME
                                        : Cell.deletionTimeUnsignedIntegerToLong((int) localDeletionTime);
```

Reply:
Will do

Reply:
Great, thanks. Just a trivial nit, there is a missed space right before the `<` operator.